src/main/java/de/natrox/common/container/Pair.java

```diff
@@ -77,4 +77,14 @@ public Pair<A, B> setSecond(B second) {
         this.second = second;
         return this;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (!obj.getClass().equals(this.getClass()))
+            return false;
+        Pair<?, ?> that = (Pair<?, ?>) obj;
+        return this.first() == that.first() && this.second() == that.second();
```

Comment:
Attention == is not the same as #equals. Since you are in an equals method, I think the values should rather be compared to #equals.

Reply:
Thanks for the help, but I wanted to wait, since the entries could be null. That will be fixed later. Same goes for your second review comment about the same mistake in Triple#equals.